data/datasets/oa_cpp_annotate/process_data.py

```diff
@@ -0,0 +1,93 @@
+# %%
+import random
+
+import pandas as pd
+
+SPLIT_1 = "Q. What language is the above code written in?\nA. "
+# SPLIT_2 = "Q. What is the purpose of the above code?\nA."
+JSONL_FP = "ghdecomp_summaries_merged_small_openaisummary_combined.jsonl.bz2"
+
+PROMPTS = [
+    "{code}What language is the above code written in? What does the code do?\n\n",
```

Comment:
The prompts which end with `\n\n` cause INSTRUCTION values ending with `\n\n` in the OA format dataset. I don't think this is desirable

Reply:
Got it. I'll remove them and reupload the data. I'm also working on inverting the task to add some in code generation instructions. Want me to add it to this dataset, or do you think it would be better to leave this as a pure annotation task?

Reply:
I think code generation tasks are in general useful. I am sceptical of "duplicating" data in training (e.g. having a Q-A pair where the model predicts code from description, and an inverted Q-A pair where the model predicts description from code, both derived from the same original description-code pair).

One approach which has been taken in the past (with poetry data and others) is to randomise this, such that half of the description-code pairs are used for one task and the other half for the other task.

Reply:
I have similar qualms about duplicated data. My plan was to random assign a sample to either the code gen or code annotation task. I'm not sure what the dev team's priorities are in terms of dataset composition. My naive idea was something like a 70-30 split between annotation and generation as the crowd-sourced chat data already has a fair amount of code gen in it. Do you have any preferences?

Reply:
> I have similar qualms about duplicated data. My plan was to random assign a sample to either the code gen or code annotation task. I'm not sure what the dev team's priorities are in terms of dataset composition. My naive idea was something like a 70-30 split between annotation and generation as the crowd-sourced chat data already has a fair amount of code gen in it. Do you have any preferences?

I will leave this up to you, it's true that there is much more code gen than annotation currently so something like the 70-30 split could make sense. It's hard to know what is "optimal" without training models to test

Reply:
I'll get to work on this. Thank you for the feedback.